optashift-employee-rostering-server/src/main/resources/org/optaplanner/openshift/employeerostering/server/solver/employeeRosteringScoreRulesTemplate.drl.vm

```diff
@@ -0,0 +1,114 @@
+#set ( $header = '/*
+ * Copyright 2010 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.openshift.employeerostering.server.solver;
+    dialect "java"
+
+import java.time.temporal.ChronoUnit;
+
+import org.optaplanner.core.api.score.buildin.hardsoft.HardSoftScoreHolder;
+import org.optaplanner.openshift.employeerostering.shared.employee.Employee;
+import org.optaplanner.openshift.employeerostering.shared.employee.EmployeeAvailability;
+import org.optaplanner.openshift.employeerostering.shared.employee.EmployeeAvailabilityState;
+import org.optaplanner.openshift.employeerostering.shared.shift.Shift;
+import org.optaplanner.openshift.employeerostering.shared.timeslot.TimeSlot;
+import org.optaplanner.openshift.employeerostering.shared.skill.Skill;
+import org.optaplanner.openshift.employeerostering.shared.roster.Roster;
+import org.optaplanner.openshift.employeerostering.shared.spot.Spot;
+
+global HardSoftScoreHolder scoreHolder;
+
+// Hard constraints
+
+rule "Required skill for a shift"
+    when
+        Shift(
+                employee != null,
+                !getEmployee().hasSkill(getSpot().getRequiredSkill()))
+    then
+        scoreHolder.addHardConstraintMatch(kcontext, -100);
```

Comment:
Use the Parameterization pattern instead of using a velocity template, see examination example in optaplanner-examples:

```
public class RosterParametrization extends AbstractPersistable {

    private int twoInARowPenalty;
    private int twoInADayPenalty;
    private int periodSpreadLength;
    private int periodSpreadPenalty;
    private int mixedDurationPenalty;
    private int frontLoadLargeTopicSize;
    private int frontLoadLastPeriodSize;
    private int frontLoadPenalty;
```

```
@PlanningSolution()
public class Roster {

    @ProblemFactProperty
    private RosterParametrization rosterParametrization;
```


```
// Exams which share students have to few periods between them
rule "periodSpread"
    when
        $rosterParametrization : RosterParametrization(periodSpreadPenalty != 0)
       ...
    then
        scoreHolder.addSoftConstraintMatch(kcontext,
               ... * (- $rosterParametrization.getPeriodSpreadPenalty()));
end
```
https://github.com/kiegroup/optaplanner/blob/master/optaplanner-examples/src/main/resources/org/optaplanner/examples/examination/solver/examinationScoreRules.drl#L160